package/MDAnalysis/coordinates/LAMMPS.py

```diff
@@ -274,46 +275,63 @@ def __init__(self, filename, convert_units=True, **kwargs):
         self.units['velocity'] = kwargs.pop('velocityunit',
                                  self.units['length']+'/'+self.units['time'])
 
-    def _write_atoms(self, atoms, data):
+    def _write_atoms(self, atoms, data, atom_style):
         self.f.write('\n')
-        self.f.write('Atoms\n')
+        self.f.write('Atoms  # ' + atom_style + '\n')
         self.f.write('\n')
 
         try:
             charges = atoms.charges
+            has_charges = True
         except (NoDataError, AttributeError):
             has_charges = False
-        else:
-            has_charges = True
 
         indices = atoms.indices + 1
         types = atoms.types.astype(np.int32)
 
-        moltags = data.get("molecule_tag", np.zeros(len(atoms), dtype=int))
+        if atom_style == 'full':
+            print_moltag = True
+            moltags = data.get("molecule_tag", np.zeros(len(atoms), dtype=int))
+            # TODO: if not exist, shall return None and not print to file, instead of 0s
+            # also may use resid instead of 'molecule_tag'
+        else:  # assume style is atomic.
+            print_moltag = False
 
         if self.convert_units:
             coordinates = self.convert_pos_to_native(atoms.positions, inplace=False)
+        else: 
+            coordinates = atoms.positions
 
-        if has_charges:
+        if has_charges and print_moltag:
```

Comment:
It may be better to separate these by `atom_style` it is not very clear to which style these branches belong (should use a `case` statement). Not really your fault but will make things a lot clearer.